api/pkg/filtermanager/filtermanager.go

```diff
@@ -248,7 +248,8 @@ func (m *filterManager) handleAction(res api.ResultAction, phase phase, filter *
 		} else if phase == phaseEncodeHeaders {
 			m.encodeResponseNeeded = true
 		} else {
-			api.LogErrorf("WaitAllData only allowed when processing headers, phase: %v", phase)
+			api.LogErrorf("WaitAllData only allowed when processing headers, phase: %v. "+
+				" If you need to buffer the body, please use DecodeRequest or EncodeResponse instead", phase)
```

Comment:
```suggestion
				" In the mean time, use DecodeRequest /  EncodeResponse instead of DecodeData / EncodeData to handle fully buffered body.", phase)
```